client/components/Assignments/AssignmentItem/fields/Name.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+interface IProps {
+    assignment: any;
+    fetchPlanningItemForAssignment: any;
+}
+
+export const NameComponent = ({
+    assignment,
+    fetchPlanningItemForAssignment,
+}: IProps) => {
+    const [name, setName] = React.useState(null);
+
+    if (name === null) {
```

Comment:
There are data syncing and performance concerns with this approach:

1. If the name of the Planning item changes, this component will not update
2. An API request is sent for every Assignment in the list view, for every session currently in the Assignment workspace (albeit on load only)

To get around this issue we decided to sync the Planning details required to the Assignment.

1. Add `name` to the Assignment schema (https://github.com/superdesk/superdesk-planning/blob/master/server/planning/assignments/assignments.py#L1241)
2. Sync `name` from the Planning item to the Assignment (https://github.com/superdesk/superdesk-planning/blob/master/server/planning/planning/planning.py#L716)
3. Finally update the mongo/elastic records for the Assignment (https://github.com/superdesk/superdesk-planning/blob/master/server/planning/planning/planning.py#L734)

Then you can access the `name` in the same way that `description_text` is being used.

Reply:
I guess I looked for where to change this on the client instead of the server... Thanks for sharing 👍